tests/functional/src/main/java/org/mule/functional/api/classloading/isolation/AetherClassPathClassifier.java

```diff
@@ -0,0 +1,692 @@
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+
+/**
+ * Creates the {@link ArtifactUrlClassification} based on the Maven dependencies declared by the rootArtifact using Eclipse
+ * Aether. Uses a {@link DependencyResolver} to resolve Maven dependencies.
+ * <p/>
+ * The classification process classifies the rootArtifact dependencies in three groups: {@code provided}, {@code compile} and
+ * {@code test} scopes. It resolves dependencies graph for each group applying filters and exclusions and classifies the list of
+ * {@link URL}s that would define each class loader container, plugins and application.
+ * <p/>
+ * Dependencies resolution uses dependencies management declared by these artifacts while resolving the dependency graph.
+ * <p/>
+ * Plugins are discovered as {@link Extension} if they do have a annotated a {@link Class}. It generates the {@link Extension}
+ * metadata in order to later register it to an {@link org.mule.runtime.extension.api.ExtensionManager}.
+ *
+ * @since 4.0
+ */
+public class AetherClassPathClassifier implements ClassPathClassifier {
+
+  public static final String POM = "pom";
+  public static final String POM_XML = POM + ".xml";
+  public static final String MAVEN_COORDINATES_SEPARATOR = ":";
+  public static final String JAR_EXTENSION = "jar";
+  public static final String SNAPSHOT_WILCARD_FILE_FILTER = "*-SNAPSHOT*.*";
+  public static final String MULE_EXTENSION = "mule-extension";
+  public static final String GENERATED_TEST_RESOURCES = "generated-test-resources";
+  public static final String TESTS_CLASSIFIER = "tests";
+
+  protected final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+  private DependencyResolver dependencyResolver;
+
+  /**
+   * Creates an instance of the classifier.
+   *
+   * @param dependencyResolver {@link DependencyResolver} to resolve dependencies
+   */
+  public AetherClassPathClassifier(DependencyResolver dependencyResolver) {
+    checkNotNull(dependencyResolver, "dependencyResolver cannot be null");
+    this.dependencyResolver = dependencyResolver;
+  }
+
+  /**
+   * Classifies {@link URL}s and {@link Dependency}s to define how the container, plugins and application class loaders should be
+   * created.
+   *
+   * @param context {@link ClassPathClassifierContext} to be used during the classification
+   * @return {@link ArtifactUrlClassification} as result with the classification
+   */
+  @Override
+  public ArtifactUrlClassification classify(ClassPathClassifierContext context) {
+    Artifact rootArtifact = getRootArtifact(context);
+    if (logger.isDebugEnabled()) {
+      logger.debug("Building class loaders for rootArtifact: {}", rootArtifact);
+    }
+    List<Dependency> directDependencies = dependencyResolver
+        .getDirectDependencies(rootArtifact);
```

Comment:
Not necessarily wrong, but why do you use a new line here? is the formatting adding it?